### PR TITLE
docs: Provide more structured API docs for Workflow API

### PR DESCRIPTION
**Pull request by tomkerkhove**
Thank you for helping make the Dapr documentation better!

**Please follow this checklist before submitting:**
- [x] Commits are signed with Developer Certificate of Origin (DCO - [learn more](https://docs.dapr.io/contributing/contributing-overview/#developer-certificate-of-origin-signing-your-work))
- [x] [Read the contribution guide](https://docs.dapr.io/contributing/docs-contrib/contributing-docs/)
- [ ] Commands include options for Linux, MacOS, and Windows within codetabs
- [ ] New file and folder names are globally unique
- [ ] Page references use shortcodes instead of markdown or URL links
- [ ] Images use HTML style and have alternative text
- [ ] Places where multiple code/command options are given have codetabs

In addition, please fill out the following to help reviewers understand this pull request:

## Description

Provide more structured API docs for Workflow API to make clear what is required and expected.

## Issue reference

N/A

### CODE REVIEW

File: daprdocs/content/en/reference/api/workflow_api.md
Comment (cgillum) [inline daprdocs/content/en/reference/api/workflow_api.md]: Calling it "arguments" (plural) isn't quite correct since workflows can only take a single argument. Also, the argument doesn't necessarily have to be a JSON object, which seems to be implied by the `{...}` here. It can be any valid JSON data, including objects, strings, numbers, or arrays.

File: daprdocs/content/en/reference/api/workflow_api.md
Comment (cgillum) [inline daprdocs/content/en/reference/api/workflow_api.md:86]: I'm just noticing that the `Content-Type` and `Content-Length` fields in the response looks wrong. Would you mind removing these for us as part of this PR?

File: daprdocs/content/en/reference/api/workflow_api.md
Comment (cgillum) [inline daprdocs/content/en/reference/api/workflow_api.md]: Can we remove this whole `## Headers` section? It's not correct. We have no such feature today.

File: daprdocs/content/en/reference/api/workflow_api.md
Comment (cgillum) [inline daprdocs/content/en/reference/api/workflow_api.md]:
```suggestion
However, Dapr comes with a built-in `dapr` workflow component that is built on Dapr Actors. No component file is required to use the built-in Dapr workflow component.
```

**Review by cgillum — COMMENTED** [review]
Thanks for this PR! A few suggestions below.

File: daprdocs/content/en/reference/api/workflow_api.md
Comment (tomkerkhove) [reply daprdocs/content/en/reference/api/workflow_api.md]: That is what I figured but kept it around, will remove it.

File: daprdocs/content/en/reference/api/workflow_api.md
Comment (tomkerkhove) [reply daprdocs/content/en/reference/api/workflow_api.md]: Updated

File: daprdocs/content/en/reference/api/workflow_api.md
Comment (tomkerkhove) [reply daprdocs/content/en/reference/api/workflow_api.md]: Updated

File: daprdocs/content/en/reference/api/workflow_api.md
Comment (tomkerkhove) [reply daprdocs/content/en/reference/api/workflow_api.md]: Updated with a proposal based on what you shared - Thoughts?

**Comment by tomkerkhove** [comment]
Good feedback and updated the PR - Let me know what you think @cgillum!

**Comment by tomkerkhove** [comment]
Oh hello my good old DCO friend.

**Review by hhunter-ms — APPROVED** [review]
lgtm - @cgillum what do you think?

File: daprdocs/content/en/reference/api/workflow_api.md
Comment (cgillum) [inline daprdocs/content/en/reference/api/workflow_api.md]:
```suggestion
Start a workflow instance with the given name and instance ID.
```

File: daprdocs/content/en/reference/api/workflow_api.md
Comment (cgillum) [inline daprdocs/content/en/reference/api/workflow_api.md]: We don't accept any specific headers. Did you mean to say something about the workflow input here?

File: daprdocs/content/en/reference/api/workflow_api.md
Comment (cgillum) [inline daprdocs/content/en/reference/api/workflow_api.md]:
```suggestion
### HTTP response codes
```

File: daprdocs/content/en/reference/api/workflow_api.md
Comment (cgillum) [inline daprdocs/content/en/reference/api/workflow_api.md]:
```suggestion
### Response content
```

File: daprdocs/content/en/reference/api/workflow_api.md
Comment (cgillum) [inline daprdocs/content/en/reference/api/workflow_api.md]:
```suggestion
Terminate a running workflow instance with the given name and instance ID.
```

File: daprdocs/content/en/reference/api/workflow_api.md
Comment (cgillum) [inline daprdocs/content/en/reference/api/workflow_api.md]:
```suggestion
### HTTP response codes
```

File: daprdocs/content/en/reference/api/workflow_api.md
Comment (cgillum) [inline daprdocs/content/en/reference/api/workflow_api.md]:
```suggestion
### Response content
```

File: daprdocs/content/en/reference/api/workflow_api.md
Comment (cgillum) [inline daprdocs/content/en/reference/api/workflow_api.md]:
```suggestion
### Request content
```

File: daprdocs/content/en/reference/api/workflow_api.md
Comment (cgillum) [inline daprdocs/content/en/reference/api/workflow_api.md]:
```suggestion
### HTTP response codes
```

File: daprdocs/content/en/reference/api/workflow_api.md
Comment (cgillum) [inline daprdocs/content/en/reference/api/workflow_api.md]:
```suggestion
### Response content
```

**Review by cgillum — COMMENTED** [review]
Oh sorry - I forgot that this PR was waiting for more feedback from me. Here are a few nitpicky things I suggest we fix before merging.

File: daprdocs/content/en/reference/api/workflow_api.md
Comment (cgillum) [reply daprdocs/content/en/reference/api/workflow_api.md]: Yeah, I like the update you provided.

File: daprdocs/content/en/reference/api/workflow_api.md
Comment (tomkerkhove) [reply daprdocs/content/en/reference/api/workflow_api.md]: I guess so, I forgot 🤷‍♂️

What about this?

```suggestion
In the request you can pass along relevant input information that will be passed to the workflow:
```

Comment (cgillum) [reply daprdocs/content/en/reference/api/workflow_api.md]: LGTM!

**Comment by cgillum** [comment]
@tomkerkhove were you planning on making that last change related to the sentence starting with "In the request you can pass along..."?

**Comment by tomkerkhove** [comment]
> @tomkerkhove were you planning on making that last change related to the sentence starting with "In the request you can pass along..."?

Yes I was, but I was sleeping :D

Done now!

**Review by cgillum — APPROVED** [review]
LGTM!

**Pull request merged**
